### PR TITLE
Add zoneless scheduler to Angular by default

### LINKED ISSUE #11565: [Bug] Animation not triggered on window.resize
**I'm submitting a ...**  (check one with "x")
```
[x] bug report => search github for a similar issue or PR before submitting
```

**Current behavior** 
I have a page with a side menu, which has an animation on it's width, the item does work when i change the state variable from a click to a button that calls a function that sets the value for the state of animation, however it doesn't happen when calling the same function from an Rxjs observable triggered from an event `onresize`.

Basically i want to always close the menu when a resize event happens, may it have to be with the Observable or the resize event?

Also worth nothing i have some primeng animations in the menu that when triggered it triggers the menu closing animation held back when resizing, which makes me think that the state binding is being held due to binding to the Observable or to a window event, then when other animation happens the change detection kicks again and the animation triggers.

Here's my observable code in the constructor:
```ts
@Component({
  selector: 'app',
  encapsulation: ViewEncapsulation.None,
  styleUrls: [
    './app.style.scss'
  ],
  animations: [
    animMenuRight
  ],
  templateUrl: './app.template.html'
})
export class App {
private resizeSub:  Subscription;
  private isMenuOpen: boolean = false;

  constructor(
    public appState: AppState
  ) {
    // this.resizeSub = Observable.fromEvent(document, 'onresize')
    // .debounceTime(250)
    // .subscribe((event) => {
    //   console.log('event: ', event);
    //   this.toggleMenu(this.isMenuOpen);
    // });

    this.resizeSub = Observable.create((observer) => {
      window.onresize = (event) => observer.next(event);
    })
    .debounceTime(250)
    .subscribe((event) => {
      this.toggleMenu(true);
    });
  }
  toggleMenu(isOpen) {
    console.log('toggleMenu: ', isOpen, window.innerWidth);

    if (window.innerWidth >= 1024) {
      return this.isMenuOpen = false;
    }
    this.isMenuOpen = isOpen;
  }
}
```

Here's the template animation binding:
```html
<l-toolbar [toggleMenuValue]="isMenuOpen" (toggleMenuEvent)="toggleMenu($event)"></l-toolbar>

<div class="app-container">

  <div [@animMenuRight]='isMenuOpen' class="sidemenu">
    <p-slideMenu [model]="routes"></p-slideMenu>
  </div>
  <div class="main-content">


    <router-outlet></router-outlet>

    <!--<l-menu-overlay></l-menu-overlay>-->

  </div>
</div>
```

And here's the animation itself:
```ts
import {
  Component,
  Input,
  trigger,
  state,
  style,
  transition,
  animate
} from '@angular/core';

export var animMenuRight = trigger('animMenuRight', [
  state('true', style({
    flex: '0 1 0px'
  })),
  state('false', style({
    flex: '0 0 150px'
  })),
  transition('* => *', animate('250ms ease'))
]);
```

**Expected behavior**
I expect to click a button and the menu should toggle between open-close, which does work ok,
i expect the resize of the window to trigger the same behaviour but it doesn't.

**Reproduction of the problem**
This is my repo, it's a public repo so it should be easy to clone and run:
https://github.com/Luchillo/Luchillo-tech-showcase

**Please tell us about your environment:**

* **Angular version:** 2.0.0-rc.6

* **Browser:** [all | Chrome 53 | Safari 9.1.1] 
I've tested in chrome 53 and Safari 9.1.1 both up to date, most likely this issue happens in all browsers
 
* **Language:** [all]

### LINKED ISSUE #13957: Async pipe doesn't read (or update) ngrx/store observables when triggered from specific places (Google API's callbacks)
**I'm submitting a ...**  (check one with "x")
```
[x] bug report => search github for a similar issue or PR before submitting
[ ] feature request
[ ] support request => Please do not submit support request here, instead see https://github.com/angular/angular/blob/master/CONTRIBUTING.md#question
```

**Current behavior**

Google API's calls take a callback, when dispatching an ngrx/store "action" from that callback, the store receives and updates the value. Even observers subscribed to that value receive it, but `async` pipes don't take the value.

But if there's an event (a click in a button) then the `async` pipes update their content.

But when called from any other place different than those callbacks (even `setTimeout`) `async` works normally.

Interestingly, when the `ngrx/store-devtools` is configured and the Redux Chrome extension installed, everything works normally.

**Expected behavior**

Whenever there's a new value in an observer, sent from any place, the `async` pipe should work and update its value.

**Minimal reproduction of the problem with instructions**

https://plnkr.co/edit/79WK9dEItgEFsHLBSN1T

*Short version*

* If you click the "Authorize" button for the first time, it triggers an authorization screen to read Gmail data using Google APIs and it stores the fact that it is authorized in `ngrx/store` as `authorized`.
* After you reload, the code tries to use the same authorization you gave and stores the result in `ngrx/store` as `authorized` (as when you clicked the button).
*  `ngrx/store` sends the new values to observers, if you open the console you will see there are observers just `console.log`ing the new values they receive.
* The new values won't show up in the page in their sections with `async` pipes.
* If you click any of the buttons in the page, that will send new actions to `ngrx/store`, but most importantly, all the places with `async` pipes will suddenly update with all the values they should have, previous and new.

**Note**: as the `console.log` subscribers do get the values and output them normally, I think this is not a [ngrx/store](https://github.com/ngrx/store) issue but an Angular one, as it seems to be related closely to the `async` pipes.

**Note**: I created a new Google API token for this Plunkr. So, when you finish testing with it, please tell me so I can disable / remove it.

*Longer version with details*

* Check in `src/app.ts` -> `ngOnInit()`: it sends an "action" to the store "on init" adding a "string" to "test", `test` in the store just adds whatever string it receives to an array.
* Check in `src/app.ts` -> `constructor()`: there's an observer for "test" values from the store.
* Check in `src/app.ts` -> in the template: There's a debug display of `test`, with an `async` pipe and a `json` pipe.
* On init, that `test` action and value shows up in the section with the `async` pipe correctly. So `async` pipes work well when the observers' values sent are triggered from normal functions.
* Check in `src/app.ts` -> `ngOnInit()`: it also triggers a `checkAuth()` from `gapiService`.
* Check in `src/gapi.service.ts` -> `checkAuth()`: that `checkAuth()` will check if it is already authorized. If it is authorized (after the first time you authorize it by clicking the button and reload the page) it will dispatch an action to `ngrx/store` stored as `authorized`. 
* Check in `src/app.ts` -> `constructor()`: there's an observer for "authorized" values from the store that just `console.log`s them.
* Check your console: there will be some `new ngrx/store.authorized value:` with the value.
* Check that the "authorized" section of the page with the `async` pipe didn't update.
* Check in `src/gapi.service.ts` -> `checkAuth()`: in the same callback that stores the `authorized` value in the `ngrx/store` is another value sent to the `ngrx/store` saved as `test`.
* Check in `src/app.ts` -> `constructor()`: there's an observer for `test` values from the store that just `console.log`s them too.
* Check your console: there will be some `new ngrx/store.test value:` with the value. So observers do work. Is just the `async` pipe that is not working as expected.
* In the page, in the "test" section, the array should show all the values sent to `ngrx/store` in `test`, but it only shows the one sent before from the `ngOnInit()`, not the ones from the Google API's callbacks.
* Check in `src/app.ts` -> the template: there is a `click` event handler for the `Test` button.
* Check in `src/app.ts` -> `test()` (the `click` handler): It just dispatches another `test` action to `ngrx/store`.
* As this action is not sent from a Google API callback, the `async` pipe gets the new value immediately, and it takes all the previous values too.
* Although the "Test" button only sends a new value to `test` in `ngrx/store`, the `async` pipe for `authorized` also updates its value and the page shows "true".

**What is the motivation / use case for changing the behavior?**

`async` pipes are supposed to subscribe to any observer and update their final value based on those observers, independent of where the update of those observers was triggered.

**Please tell us about your environment:**

OS: 
Distributor ID: Ubuntu
Description:    Ubuntu 16.04.1 LTS
Release:        16.04
Codename:       xenial

Browser: Chrome 54.0.x (64 bit) and Firefox 50.1.0

IDE: Atom with `atom-typescript` and Visual Studio Code

`npm version`: 

```
{ 'gapi-test': '0.0.0',
  npm: '3.10.8',
  ares: '1.10.1-DEV',
  http_parser: '2.7.0',
  icu: '56.1',
  modules: '48',
  node: '6.9.1',
  openssl: '1.0.2j',
  uv: '1.9.1',
  v8: '5.1.281.84',
  zlib: '1.2.8' }
```

`angular-cli` version (but the Plunkr version doesn't use `angular-cli` and the issue still occurs):

```
angular-cli: 1.0.0-beta.25.5
node: 6.9.1
os: linux x64
@angular/common: 2.4.3
@angular/compiler: 2.4.3
@angular/core: 2.4.3
@angular/forms: 2.4.3
@angular/http: 2.4.3
@angular/platform-browser: 2.4.3
@angular/platform-browser-dynamic: 2.4.3
@angular/router: 3.4.3
@angular/compiler-cli: 2.4.3
```

yarn version: `yarn version v0.17.8`

`@angular/core` (taken from `yarn.lock`): `"@angular/core@^2.3.1": version "2.4.3"`

* **Angular version:** 2.0.X

* **Browser:** [ Chrome XX | Firefox XX | Mobile Chrome XX ]

 
* **Language:** [ typescript@2.0.2 (in Plunkr) | typescript@~2.0.3 (yarn version "2.0.10") (in local angular-cli project ]

* **Node (for AoT issues):** `node --version` =   v6.9.1

### LINKED ISSUE #15770: Router: change detection doesn't work with manual initial navigation (is it a bug?)
**I'm submitting a ...** (check one with "x")

```
[ x ] bug report => search github for a similar issue or PR before submitting
[ ] feature request
[ ] support request => Please do not submit support request here, instead see https://github.com/angular/angular/blob/master/CONTRIBUTING.md#question
```

**Current behavior**
Change detection doesn't work in initially loaded components if the router was configured to perform a manual initial navigation (`{ initialNavigation: false }` and `router.initialNavigation()`).
If navigation was triggered by the link like _\<a routerLink=...\>_, change detection works fine.

**Expected behavior**
Change detection works normally in both cases.

**_Note:_** after some debugging I figured out that if `router.initialNavigation()` is called within `ngZone.run()`, change detection works. But it is not a straightforward behavior.

**Minimal reproduction of the problem with instructions**

Repository with reproduction: [https://github.com/n9niwas/issues-ng4-router.git](https://github.com/n9niwas/issues-ng4-router.git).

1. open fresh browser window and navigate to _\/test/_;
1. property "x" is not being updated in the view on `setInterval()` ticks.

**What is the motivation / use case for changing the behavior?**

Initial navigation has to be performed manually in hybrid applications because there is no bootstrapping (empty `ngDoBoostrap()`).

**Please tell us about your environment:**

Operating system: Windows 10.
IDE: VS Code.
Package manager: yarn.

* **Angular version:** 4.0.1

* **Browser:** [**all** | Chrome XX | Firefox XX | IE XX | Safari XX | Mobile Chrome XX | Android X.X Web Browser | iOS XX Safari | iOS XX UIWebView | iOS XX WKWebView ]

* **Language:** [all | **TypeScript 2.2.2** | ES6/7 | ES5]

### LINKED ISSUE #15946: Router can't navigate when exception occurrs
**I'm submitting a ...**  (check one with "x")
```
[V ] bug report => search github for a similar issue or PR before submitting

```

**Current behavior**

Router.Navigate fails to navigate when exception occurs
**Expected behavior**

Router is  able to navigate even though application exception occurred
**Minimal reproduction of the problem with instructions**

I have created a plunker here, you can comment out line 16 of ContactComponent to see the no exception scenario, and enable it to see the error scenario. I purposely created this exception to test out exception navigation, I expect my custom error handler GlobalErrorService is able to navigate the page to ErrorComponent, but that's not happening

https://embed.plnkr.co/HhYYHnqAOHzFAvKc7o58/

**What is the motivation / use case for changing the behavior?**

**Please tell us about your environment:**

Windows 

* **Angular version:** 4.0.2

* **Browser:** [all | Chrome XX | Firefox XX | IE XX | Safari XX | Mobile Chrome XX | Android X.X Web Browser | iOS XX Safari | iOS XX UIWebView | iOS XX WKWebView ]

 
* **Language:** [all | TypeScript X.X | ES6/7 | ES5]

* **Node (for AoT issues):** `node --version` =

### LINKED ISSUE #18254: Using router.navigate  to navigate to another component does not invoke the onInit method
Hi I am implementing a project in angular . and i am using the version ng-version="4.3.0" 
Here i am redirecting the app to another component doesnt invoking the oninit  method

Here is my sample code 
```
`fblogin() {
        const provider = 'facebook';
        this.socialloginservice.onFacebookLogin().subscribe(
            response => {
                this.authService.socialSignup(provider, response).subscribe(res => {
                    console.log(this.authService.getToken());
                    console.log('redirecting');
                    this._router.navigate(['/register-step2']);
                });

            },
            error => console.log(error)
        );
    }` 

```

I am redirecting the app to another route in the success of an http service call
## Current behavior
In Register step2 component
```
constructor(private authenticateservice: AuthenticateService,
              ) {
     console.log('ok Working');

  }

ngOnInit() {
 //not working
}
```

## Expected behavior

```
constructor(private authenticateservice: AuthenticateService,
              ) {
     console.log('ok Working');

  }

ngOnInit() {
 //should work
}

```

i have checked for this issue in the web and i found the articles like below saying to add below scripts
<script src="node_modules/angular2/bundles/angular2-polyfills.js"></script>
<script src="node_modules/systemjs/dist/system.js"></script>

but i didn't have such folders in my node modules. i am not getting where to import these files and from where to import.
Please let me know where i am missing
**https://github.com/angular/angular/issues/4809**

## Environment

<pre><code>
Angular version: 4.3.0

Browser:
- [ ] Chrome (desktop) latest version

### LINKED ISSUE #19731: Angular freezes without any error when I navigate to next page from an external library callback
## I'm submitting a...

<pre><code>
[ ] Regression (a behavior that used to work and stopped working in a new release)
[x] Bug report  
[ ] Feature request
[ ] Documentation issue or request
[ ] Support request => Please do not submit support request here, instead see https://github.com/angular/angular/blob/master/CONTRIBUTING.md#question
</code></pre>

## Current behavior

There are 2 components: Login and Main page. I use Google JS API `gapi` to login. If I navigate to main page once I have logged in successfully from Google, then in Main page it freezes. Yet if I go to Main page without logging in first then main page works. And then if you go back to login page and login with Google api, it redirect to main page with works fine too.

## Expected behavior

It should works consistently if I navigate to another page from an external callback.

## Minimal reproduction of the problem with instructions

https://plnkr.co/edit/8IJJReAwtAOk8GE3enuS?p=preview

## What is the motivation / use case for changing the behavior?

I want users to login first before navigating to other pages but all broken

## Environment

<pre><code>
Angular version: 4.4.5

Browser:
- [61] Chrome (desktop) version XX
- [ ] Chrome (Android) version XX
- [ ] Chrome (iOS) version XX
- [ ] Firefox version XX
- [ ] Safari (desktop) version XX
- [ ] Safari (iOS) version XX
- [ ] IE version XX
- [ ] Edge version XX
 
For Tooling issues:
- Node version: XX  
- Platform:  

Others:

</code></pre>

### LINKED ISSUE #19814: Change detection is not updating the view when you mark for check.
## I'm submitting a...

<pre><code>
[ ] Regression (a behavior that used to work and stopped working in a new release)
[x] Bug report  
[ ] Feature request
[ ] Documentation issue or request
[ ] Support request => Please do not submit support request here, instead see https://github.com/angular/angular/blob/master/CONTRIBUTING.md#question
</code></pre>

## Current behavior

When change detector is marked for check after an external http response(in my case a Facebook SDK callback) continued by a timeout(or a new angular http subscription), the view is not updated with the new values.

If any other event is triggered in browser after that, the view is being updated with new values.

**Note 1**: If we remove the timeout from `setTimeout`(or set a lower value, like `1`), the view will be updated as expected.

**Note 2**: It is the same even if we remove the `ChangeDetectionStrategy.OnPush` strategy.

## Expected behavior

The view should be updated with the new values when we mark for check.

## Minimal reproduction of the problem with instructions

http://plnkr.co/edit/aPu9e1zwUxSAGVScLYvq?p=preview

1. Click the "Set user" button ==>> User is not displayed after 1 second.
2. Click the "Fire an event" button ==>> User is displayed.

## What is the motivation / use case for changing the behavior?

The view is not updated properly. Have to wait for the next event.

## Environment

<pre><code>
Angular version: 4.4.6

Browser:
- [x] Chrome (desktop) version Version 61.0.3163.100 (Official Build) (64-bit)
- [ ] Chrome (Android) version XX
- [ ] Chrome (iOS) version XX
- [ ] Firefox version XX
- [ ] Safari (desktop) version XX
- [ ] Safari (iOS) version XX
- [ ] IE version XX
- [ ] Edge version XX
 
For Tooling issues:
- Node version: 6.11.2  
- Platform:  Mac

Others:

</code></pre>

### LINKED ISSUE #20112: Angular 5 & 4.4.x - Router Navigate does not activate onInit after external request
<pre><code>
[ ] Regression (a behavior that used to work and stopped working in a new release)
[X] Bug report  
[ ] Feature request
[ ] Documentation issue or request
[ ] Support request => Please do not submit support request here, instead see https://github.com/angular/angular/blob/master/CONTRIBUTING.md#question
</code></pre>

## Current behavior
On external javascript API response, (created as Observable) the router.navigate method, does not activate lifecycle events "OnInit" or "ngAfterViewInit"

## Expected behavior
After a response from external JS API, like Facebook, Gigya, etc. And navigate to route, the lifecycle should be triggered.

## Minimal reproduction of the problem with instructions
This is a Router issue, please go to next repo with minimal environment:

[https://github.com/elporfirio/angular4-router-issue](https://github.com/elporfirio/angular4-router-issue)

1. Install and ng-serve
2. Enter to:  http://localhost:4200/
3. Press Click me ! (for xhr request) and everything works good (you should see 2 alerts)
4. Press Click me for Error (for api js request) and lyfecycle stop working

In file [service HelperService](https://github.com/elporfirio/angular4-router-issue/blob/master/src/app/helper-service.service.ts) exist 2 methods

First One a simple XHR works fine, but the other one use a JS API after complete a request (success/error no matter). The navigate redirects but the lifecycle is not activated.

## What is the motivation / use case for changing the behavior?
After navigate the component should work normally, as on type url.

## Environment

<pre><code>
Angular version: 5.0  and 4.x

    "@angular/animations": "^5.0.0",
    "@angular/common": "^5.0.0",
    "@angular/compiler": "^5.0.0",
    "@angular/core": "^5.0.0",
    "@angular/forms": "^5.0.0",
    "@angular/http": "^5.0.0",
    "@angular/platform-browser": "^5.0.0",
    "@angular/platform-browser-dynamic": "^5.0.0",
    "@angular/router": "^5.0.0",
    "core-js": "^2.4.1",
    "rxjs": "^5.5.2",
    "zone.js": "^0.8.14"

Browser:
- [X] Chrome (desktop) Versión 61.0.3163.100 (Build oficial) (64 bits)
- [X] Safari (desktop) Versión 11.0.1 (12604.3.5.1.1)
 
For Tooling issues:
- Node version: v6.11.3
- Platform:  MacOS Sierra

</code></pre>

### LINKED ISSUE #22472: On navigate() router method ngOnInit of target component is not called
## I'm submitting a...
<pre><code>
[ ] Regression (a behavior that used to work and stopped working in a new release)
[X ] Bug report  
[ ] Feature request
[ ] Documentation issue or request
[ ] Support request => Please do not submit support request here, instead see https://github.com/angular/angular/blob/master/CONTRIBUTING.md#question
</code></pre>

## Current behavior
I making a redirect with this.router.navigate(['some', 'route']) ngOnInit is not called in navigated component. If I do this.router.navigateByUrl('/some/route')  everything works fine.

## Expected behavior
When doing this.router.navigate(['some', 'route']) ngOnInit in navigated component will fire.

## Environment

<pre><code>
Angular version: 5.2.5

Browser:
- [X ] Chrome (desktop) version 64.0.3282.186
- [ ] Chrome (Android) version XX
- [ ] Chrome (iOS) version XX
- [ ] Firefox version XX
- [ ] Safari (desktop) version XX
- [ ] Safari (iOS) version XX
- [ ] IE version XX
- [ X] Edge version 41.16299.248.0
 
For Tooling issues:
- Node version: XX  
- Platform:  

Others:

</code></pre>

### LINKED ISSUE #23697: Change detection is not working with router navigation
## I'm submitting a...

<pre><code>
[ ] Regression (a behavior that used to work and stopped working in a new release)
[x] Bug report  
[ ] Performance issue
</code></pre>

## Current behavior
Changedetection is not working in navigated module if routed from observable map

## Expected behavior
Changes made in navigated module should be detected in its child components

## Minimal reproduction of the problem with instructions
I created a demo app to showcase issue. 
Please see the issue #23720 for the repro in stackblitz (https://stackblitz.com/edit/cdissue)
Please download it from https://github.com/jyosh/ngIssue
Do npm install
Install angular/cli  globally if not present (npm install -g @angular/cli)
ng serve to run the app, (in case issue is not reproducible in dev build, try ng serve --prod)

The home page has two buttons - 'Go to module2 without login' & 'Go to module2 after login with gmail'

Click the first button to go to module2 page where 2 buttons (Go back,  Increment) & a count are there.
Click on increment button will increment the count. It is the working fine scenario.

Next go back and click on the button  'Go to module2 after login with gmail' which will show a popup to login through gmail. After login it will automatically route to module2 page.
You may use the credentials cdissue@gmail.com & `@ngularTest` to login using gmail

Now the count is not displayed or click on increment doesn't show the count and it is the error scenario

## What is the motivation / use case for changing the behavior?
Changedetection should work as expected even with programmatic routing

## Environment
<pre><code>
Angular version: 5.2.6

Browser:
- [x] Chrome (desktop) version XX
- [ ] Chrome (Android) version XX
- [ ] Chrome (iOS) version XX
- [ ] Firefox version XX
- [ ] Safari (desktop) version XX
- [ ] Safari (iOS) version XX
- [ ] IE version XX
- [ ] Edge version XX
</code></pre>

### LINKED ISSUE #55238: angular signal not updating when using afterNextRender()
### Which @angular/* package(s) are the source of the bug?

core, zone.js

### Is this a regression?

No

### Description

I have to manually trigger a change detection when updating/setting a signal inside the `afterNextRender` method.

```typescript
    afterNextRender(() => {      
       this.someAPI.then((items) => {
         this.items.set(items); // <-- set signal
         this.ref.detectChanges(); // <--- trigger change detection
       });       
    });
```

```html
            <tbody>
              @for(item of items(); track item.id) {
                <tr>
                  <td class="text-success"><strong>{{item.name}}</strong></td>
                </tr>
              }
            </tbody>
```

There is no mention in the documentation that this function runs outside zone.js.<br/>
So I am not sure if this is the way it is suppose to run.

### Please provide a link to a minimal reproduction of the bug

https://stackoverflow.com/questions/78280552/angular-signal-not-updating-afternextrender

### Please provide the exception or error you saw

```true
No error
```

### Please provide the environment you discovered this bug in (run `ng version`)

```true
Angular: 17.2
Node: 18
Package manager: yarn
Os: Linux
```

### Anything else?

none

### PR DESCRIPTION
**Pull request by atscott**
[fix(core): Angular should not ignore changes that happen outside the zone](https://github.com/angular/angular/commit/24d8feb17779408b31bb6401a809eb470bdf6b4e) 

When Angular receives a clear indication that change detection should
run again, this should not be ignored, regardless of what Zone it
happened in. This change updates the default change detection scheduling
approach of Zone-based applications to ensure a change detection will
run when these events happen outside the Angular zone (which includes,
for example, updating a signal that's read in a template, setting an
input of a `ComponentRef`, attaching a view marked for check, calling
`ChangeDetectorRef.markForCheck`, etc.).

This does not apply to applications using `NoopNgZone` or those which
have a custom `NgZone` implementation without ZoneJS.

The impact of this change will most often be seen in existing unit tests. Tests
execute outside the Angular Zone and this can mean that state in the
test is not fully recognized by Angular. Now that Angular will ensure
change detection _does_ run, even when the state update originates from
outside the zone, tests may observe additional rounds of change
detection compared to the previous behavior. Often, this should be seen
as more correct and the test should be updated, but in cases where it is
too much effort to debug, the test can revert to the old behavior by adding
`provideZoneChangeDetection({schedulingMode: NgZoneSchedulingMode.NgZoneOnly})`
to the `TestBed` providers.

fixes #55238
fixes https://github.com/angular/angular/issues/53844
fixes https://github.com/angular/angular/issues/53841
fixes https://github.com/angular/angular/issues/52610
fixes https://github.com/angular/angular/issues/53566
fixes https://github.com/angular/angular/issues/52940
fixes https://github.com/angular/angular/issues/51970
fixes https://github.com/angular/angular/issues/51768
fixes https://github.com/angular/angular/issues/50702
fixes https://github.com/angular/angular/issues/50259
fixes https://github.com/angular/angular/issues/50266
fixes https://github.com/angular/angular/issues/50160
fixes https://github.com/angular/angular/issues/49940
fixes https://github.com/angular/angular/issues/49398
fixes https://github.com/angular/angular/issues/48890
fixes https://github.com/angular/angular/issues/48608
fixes https://github.com/angular/angular/issues/45105
fixes https://github.com/angular/angular/issues/42241
fixes https://github.com/angular/angular/issues/41553
fixes https://github.com/angular/angular/issues/37223
fixes https://github.com/angular/angular/issues/37062
fixes https://github.com/angular/angular/issues/35579
fixes https://github.com/angular/angular/issues/31695
fixes https://github.com/angular/angular/issues/24728
fixes https://github.com/angular/angular/issues/23697
fixes https://github.com/angular/angular/issues/19814
fixes https://github.com/angular/angular/issues/13957
fixes https://github.com/angular/angular/issues/11565
fixes https://github.com/angular/angular/issues/15770
fixes https://github.com/angular/angular/issues/15946
fixes https://github.com/angular/angular/issues/18254
fixes https://github.com/angular/angular/issues/19731
fixes https://github.com/angular/angular/issues/20112
fixes https://github.com/angular/angular/issues/22472
fixes https://github.com/angular/angular/issues/23697
fixes https://github.com/angular/angular/issues/24727
fixes https://github.com/angular/angular/issues/47236

BREAKING CHANGE:

Angular will ensure change detection runs, even when the state update originates from
outside the zone, tests may observe additional rounds of change
detection compared to the previous behavior.

This change will be more likely to impact existing unit tests.
This should usually be seen as more correct and the test should be updated,
but in cases where it is too much effort to debug, the test can revert to the old behavior by adding
`provideZoneChangeDetection({schedulingMode: NgZoneSchedulingMode.NgZoneOnly})`
to the `TestBed` providers.

Similarly, applications which may want to update state outside the zone
and _not_ trigger change detection can add
`provideZoneChangeDetection({schedulingMode: NgZoneSchedulingMode.NgZoneOnly})`
to the providers in `bootstrapApplication` or add
`schedulingMode: NgZoneSchedulingMode.NgZoneOnly` to the
`BootstrapOptions` of `bootstrapModule`.

### CODE REVIEW

File: packages/core/src/application/application_ref.ts [hunk 1]
```diff
@@ -143,6 +143,16 @@ export interface BootstrapOptions {
    *
    */
   ngZoneRunCoalescing?: boolean;
+
+  /**
+   * When true, changes outside the Angular zone will _not_ schedule change detection.
+   * 
+   * When false or not provided, change detection is scheduled when Angual receives
```
Comment (eneajaho) [inline packages/core/src/application/application_ref.ts:150]:
```suggestion
   * When false or not provided, change detection is scheduled when Angular receives
```

File: packages/core/src/change_detection/scheduling/ng_zone_scheduling.ts [hunk 2]
```diff
@@ -205,6 +177,16 @@ export interface NgZoneOptions {
    *
    */
   runCoalescing?: boolean;
+  
+  /**
+   * When true, changes outside the Angular zone will _not_ schedule change detection.
+   * 
+   * When false or not provided, change detection is scheduled when Angual receives
```
Comment (eneajaho) [inline packages/core/src/change_detection/scheduling/ng_zone_scheduling.ts:184]:
```suggestion
   * When false or not provided, change detection is scheduled when Angular receives
```

File: packages/core/src/change_detection/scheduling/flags.ts [hunk 3]
```diff
@@ -0,0 +1,10 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+/** Flag to enable/disable the zoneless scheduler as default provider with zone scheduling. Th */
```
Comment (Totati) [inline packages/core/src/change_detection/scheduling/flags.ts:9]:
```suggestion
/** Flag to enable/disable the zoneless scheduler as default provider with zone scheduling. */
```
🙈

**Comment by atscott** [comment]
caretaker note: presubmit is using [this patch](http://cl/625488741), which will need to be submitted once this PR is merged.

**Review by alxhub — APPROVED** [review]

**Comment by alxhub** [comment]
This PR was merged into the repository by commit de7447d15ed964ae26f0dace4cb3b08f5cccb1c1.

**Comment by Platonn** [comment]
At the first sight I was very confused with the description of this PR - I understood that this PR would be a huuuuge breaking change denying the sense of `runOutsideAngular()`! I understood that since ng18 the change detection would be triggered after **every event/API patched by Zone.js** (i.e. setTImeout, mouse move, etc), **even if** this event/API happened in my code that I deliberately run in the `runOutsideAngular()` - to avoid change detection.

Fortunately, I'm calmed down now... 😌 after I came across @JeanMeche's great article https://riegler.fr/blog/2024-04-17-zoneless-with-zoneless-hybrid where he explains why I my understanding of this PR was wrong.

**Pull request closed without merging**